### DISCUSSION TITLE
Using typeconverters

### DISCUSSION THREAD

**Discussion by RS2009** [simolus3/drift#2774] [Q&A]
I want to store Datetime as int so that i can get precission upto milliseconds and i have seen this
https://github.com/simolus3/drift/issues/215#issuecomment-547832756
 mentioning typeconverters can you explain it more or give any other suggestions?

**Comment by simolus3** [comment]
The principle of that comment still applies, but the API has become a bit simpler in the last four years. Now, you could write that converter as

```dart
class MillisDateConverter extends TypeConverter<DateTime, int> {
  const MillisDateConverter();
  @override
  DateTime fromSql(int fromDb) {
    return DateTime.fromMillisecondsSinceEpoch(fromDb);
  }

  @override
  int toSql(DateTime value) {
    return value.millisecondsSinceEpoch;
  }
}
```

To apply it, you'd use `map` in the definition of a column, e.g.

```dart
  IntColumn get myDateTime => integer().map(const MillisDateConverter())()
```

You can read more about type converters in [the documentation](https://drift.simonbinder.eu/docs/advanced-features/type_converters/).

But if having millisecond accuracy is the only reason for introducing the type converter, consider using drift's support for [storing date times as text](https://drift.simonbinder.eu/docs/getting-started/advanced_dart_tables/#datetime-options), which supports that as well.
